pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Register the metrics.
+func Register() {
+	prometheus.MustRegister(RegistryAPIRequests)
+}
+
+const (
+	registryNamespace = "openshift"
+	registrySubsystem = "registry"
+)
+
+var (
+	RegistryAPIRequests = prometheus.NewSummaryVec(
```

Comment:
Instead of having another global variable which is always set even when not needed, can we initialize this in in `dockerregistry.Execute()` and pass it in context as well?